lib/packwerk/reference_extractor.rb

```diff
@@ -94,21 +101,21 @@ def self.get_fully_qualified_references_and_offenses_from(unresolved_references_
         constant_name: String,
         node: Parser::AST::Node,
         ancestors: T::Array[Parser::AST::Node],
-        file_path: String
+        absolute_file_path: String
       ).returns(T.nilable(UnresolvedReference))
     end
-    def reference_from_constant(constant_name, node:, ancestors:, file_path:)
+    def reference_from_constant(constant_name, node:, ancestors:, absolute_file_path:)
       namespace_path = Node.enclosing_namespace_path(node, ancestors: ancestors)
 
       return if local_reference?(constant_name, Node.name_location(node), namespace_path)
 
-      relative_path = Pathname.new(file_path).relative_path_from(@root_path).to_s
+      relative_file_path = Pathname.new(absolute_file_path).relative_path_from(@root_path).to_s
       location = Node.location(node)
 
       UnresolvedReference.new(
```

Comment:
Note that I did not change `UnresolvedReference`'s internal `Struct` implementation to name the struct variable differently. This would not only break the cache for users, but I think it would actually raise a slightly obscure error. It would be easily resolved with `rm -rf path/to/cache`, but we shouldn't break public API here. If we want to change the implementation (not worth it IMO right now), we could bust the cache if we fail to hydrate a `UnresolvedReference`.